presto-main/src/main/java/com/facebook/presto/metadata/DiscoveryNodeManager.java

```diff
@@ -105,7 +164,22 @@ private synchronized PrestoNode refreshNodesInternal()
             URI uri = getHttpUri(service);
             NodeVersion nodeVersion = getNodeVersion(service);
             if (uri != null && nodeVersion != null) {
-                PrestoNode node = new PrestoNode(service.getNodeId(), uri, nodeVersion);
+                boolean isActive = isActive(nodeVersion);
```

Comment:
Instead of changing `isActive` to take a `NodeVersion` I would rename it to `getNodeState(PrestoNode node)` and move all the below logic for figuring out the state into that function